modules/apps/commerce/commerce-checkout-web/src/main/java/com/liferay/commerce/checkout/web/internal/portlet/action/CommerceCheckoutConfigurationAction.java

```diff
@@ -0,0 +1,113 @@
+import com.liferay.commerce.util.CommerceCheckoutStepRegistry;
+import com.liferay.portal.configuration.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.ConfigurationAction;
+import com.liferay.portal.kernel.portlet.DefaultConfigurationAction;
+import com.liferay.portal.kernel.util.Constants;
+import com.liferay.portal.kernel.util.JavaConstants;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+import javax.portlet.PortletConfig;
+import javax.portlet.PortletPreferences;
+import javax.portlet.PortletRequest;
+import javax.portlet.PortletResponse;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Balázs Breier
+ */
+@Component(
+	property = "javax.portlet.name=" + CommercePortletKeys.COMMERCE_CHECKOUT,
+	service = ConfigurationAction.class
+)
+public class CommerceCheckoutConfigurationAction
+	extends DefaultConfigurationAction {
+
+	@Override
+	public String getJspPath(HttpServletRequest httpServletRequest) {
+		return "/configuration.jsp";
+	}
+
+	@Override
+	public void include(
+			PortletConfig portletConfig, HttpServletRequest httpServletRequest,
+			HttpServletResponse httpServletResponse)
+		throws Exception {
+
+		try {
+			CheckoutDisplayContext checkoutDisplayContext =
+				new CheckoutDisplayContext(
+					_commerceCheckoutStepRegistry, _configurationProvider,
+					_portal.getLiferayPortletRequest(
+						(PortletRequest)httpServletRequest.getAttribute(
+							JavaConstants.JAVAX_PORTLET_REQUEST)),
+					_portal.getLiferayPortletResponse(
+						(PortletResponse)httpServletRequest.getAttribute(
+							JavaConstants.JAVAX_PORTLET_RESPONSE)),
+					_portal);
+
+			httpServletRequest.setAttribute(
+				WebKeys.PORTLET_DISPLAY_CONTEXT, checkoutDisplayContext);
+		}
+		catch (Exception exception) {
+			_log.error(exception);
+		}
+
+		super.include(portletConfig, httpServletRequest, httpServletResponse);
+	}
+
+	@Override
+	public void processAction(
+			PortletConfig portletConfig, ActionRequest actionRequest,
+			ActionResponse actionResponse)
+		throws Exception {
+
+		String cmd = ParamUtil.getString(actionRequest, Constants.CMD);
+
+		if (cmd.equals(Constants.UPDATE)) {
+			PortletPreferences portletPreferences =
+				actionRequest.getPreferences();
+
+			portletPreferences.setValue(
+				"showCommerceCheckoutFullAddress",
+				getParameter(actionRequest, "showCommerceCheckoutFullAddress"));
+			portletPreferences.setValue(
+				"showCommerceCheckoutPhoneNumber",
+				getParameter(actionRequest, "showCommerceCheckoutPhoneNumber"));
```

Comment:
I think these should use the new name. (I don't know how they can work like this becase the save of the configuration should post different names) 🤔